tracer/src/Datadog.Trace/ClrProfiler/AutoInstrumentation/Testing/XUnit/XUnitTestAssemblyRunnerRunAsyncIntegration.cs

```diff
@@ -0,0 +1,114 @@
+// <copyright file="XUnitTestAssemblyRunnerRunAsyncIntegration.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.ComponentModel;
+using System.Reflection;
+using Datadog.Trace.Ci;
+using Datadog.Trace.ClrProfiler.CallTarget;
+using Datadog.Trace.DuckTyping;
+
+namespace Datadog.Trace.ClrProfiler.AutoInstrumentation.Testing.XUnit;
+
+/// <summary>
+/// Xunit.Sdk.TestAssemblyRunner`1.RunTestCollectionAsync calltarget instrumentation
+/// </summary>
+[InstrumentMethod(
+    AssemblyNames = new[] { "xunit.execution.dotnet", "xunit.execution.desktop" },
+    TypeName = "Xunit.Sdk.TestAssemblyRunner`1",
+    MethodName = "RunAsync",
+    ReturnTypeName = "System.Threading.Tasks.Task`1<Xunit.Sdk.RunSummary>",
+    ParameterTypeNames = new string[0],
+    MinimumVersion = "2.2.0",
+    MaximumVersion = "2.*.*",
+    IntegrationName = XUnitIntegration.IntegrationName)]
+[Browsable(false)]
+[EditorBrowsable(EditorBrowsableState.Never)]
+public static class XUnitTestAssemblyRunnerRunAsyncIntegration
+{
+    /// <summary>
+    /// OnMethodBegin callback
+    /// </summary>
+    /// <typeparam name="TTarget">Type of the target</typeparam>
+    /// <param name="instance">Instance value, aka `this` of the instrumented method.</param>
+    /// <returns>Calltarget state value</returns>
+    internal static CallTargetState OnMethodBegin<TTarget>(TTarget instance)
+    {
+        if (!XUnitIntegration.IsEnabled || instance is null)
+        {
+            return CallTargetState.GetDefault();
+        }
+
+        var assemblyRunnerInstance = instance.DuckCast<TestAssemblyRunnerStruct>();
+        if (assemblyRunnerInstance.TestAssembly.Assembly.Name is { } assemblyName)
+        {
+            var testBundleString = new AssemblyName(assemblyName).Name ?? string.Empty;
+
+            // Extract the version of the framework from the TestClassRunner base class
+            var frameworkType = instance.GetType();
```

Comment:
Just to double check, will instance have the original type here and not a proxy type (which implements `IDuckType`) here?

Reply:
that's right, `instance` is not ducktyped (the method doesn't have any constraint) the one ducktyped is `assemblyRunnerInstance`